pretyping/detyping.ml

```diff
@@ -589,8 +589,23 @@ let detype_cofix detype avoid env sigma n (names,tys,bodies) =
        Array.map (fun (_,_,ty) -> ty) v,
        Array.map (fun (_,bd,_) -> bd) v)
 
+(* TODO use some algebraic type with a case for unnamed univs so we
+   can cleanly detype them. NB: this corresponds to a hack in
+   Pretyping.interp_universe_level_name to convert Foo.xx strings into
+   universes. *)
```

Comment:
Indeed, both sides of this hack were made so that pretype (detype c) = c w.r.t universes, AFAIR. This was needed for funind but is not a bad property to have in general.